### ISSUE TITLE
Fix WCAG issues

### ISSUE THREAD

**Issue by jolevesq** [fgpv-vpgf/contributed-plugins#147]
Change slider handle to #929292 (range-slider - chart)
Draw, edit geometry not available on keyboard click, show a dialog box to explain that it is not avaialble so delete and recreate
Draw, add visual effect to the delete box
Chart, make the focus works. See what is best.... focus trap in chart and back to map... once graph close use can go to detail...

Check if we do all of them in package or one issue for each plugin.

**Comment by jolevesq** [comment]
- [ ] Range Slider handle part of #167

**Comment by jolevesq** [comment]
See Google sheet for more info
Add WCAG issue for viewer in viewer FGPV repo

WCAG SC | LEVEL | FAIL OR Best Pract. | Content Type | Template/CSS? | Component | Creative vs. Dev | Skill orEffort to Fix | CanAdapt Fixed Status: 0 fixed 0 Partially Fixed 0 Not Fixed 0 N/A 16 Total | C | Issue Name | More Description | Implications | Example URL | Example Screen ShotTo see larger image  press CTL+, or make row taller | Additional images or comments | Solution | Current Code | Solution Code Example | Comments
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1.1.1 Non-text Content | A | Fail | Button |   | Yes |   | Easy |   |   | Button or interactive control not labelled |   | Blind people will not know what it is. | https://jolevesq.github.io/contributed-plugins/thematic-slider/samples/thematic-slider-index.html?sample=99 |   |   | Give the button an aria-label |   |   |  
1.1.1 Non-text Content | A | Fail | Images |   |   |   | Moderate |   |   | graph missing alt text | graphics contain a lot of information. Simple alt text or no alt text is insufficient. and because these graphs are in HTML CANVAS, there is no real way to directly make them accessible. The data would need to be provided in a table, which I understand is not possible because it is 3rd party data that is in multiple formats. | A screen reader user who is blind will not likely be able to find out what is in the image | https://jolevesq.github.io/contributed-plugins/draw/samples/draw-index.html |   |   | There is no immediate solution to this because the data doesn't belong to the mapping team and it is not formalized in a way that could be presented to screen reader. |   |   |  
1.1.1 Non-text Content | A | Fail | Images |   |   |   | Moderate |   |   | graph missing alt text | graphics contain a lot of information. Simple alt text or no alt text is insufficient. and because these graphs are in HTML CANVAS, there is no real way to directly make them accessible. The data would need to be provided in a table, which I understand is not possible because it is 3rd party data that is in multiple formats. | A screen reader user who is blind will not likely be able to find out what is in the image | https://jolevesq.github.io/contributed-plugins/chart/samples/chart-index.html?sample=5 |   |   | There is no immediate solution to this because the data doesn't belong to the mapping team and it is not formalized in a way that could be presented to screen reader. |   |   |  
1.3.1 Info and Relationships | A | Fail |   |   |   |   |   |   |   | The side panel of data is visually structured but is not semantically structured. |   |   | https://jolevesq.github.io/contributed-plugins/chart/samples/chart-index.html?sample=55 |   |   | Add a heading 4 and definition lists for the JSON type relationships. CSS can be used to present the dt and dd on the same line |   |   |  
1.4.11 Contrast (Minimum) | AA | Fail | Colour or Contrast |   |   |   | Moderate |   |   | Button that is not disabled doesn't have a color contrast of 3:1 with the background |   | People with low vision and colour blindness may have trouble seeing it. It's difficult to see content that doesn't have sufficient contrast. | https://jolevesq.github.io/contributed-plugins/range-slider/samples/range-slider-index.html?sample=14 |   |   | Solution 1: Make the button darker. If it was a clear white background it would need to be at least #929292, but since its not a clear background, but has some transparency, it shpuld be darker. Make it at least #757575 |   |   | A global strategy to address the colour contrast issues may be to provide a high contrast icon on the top right corner of the site which will make the site easier to use for people with low vision. A free contrast tool can be downloaded. Google "Color Contrast Tool" (British spelling of color)
1.4.11 Non-Text Contrast | AA | Fail | Interface contrast |   |   |   | Easy |   |   | Interactive controls don't have a border or any identifier that passes the minimum contrast requiremnets of 3:1 with the background |   |   |   |   |   | Solution 1: Make the buttons and icons darker. If it was a clear white background it would need to be at least #929292, but since its not a clear background, but has some transparency, it should be darker. Make it at least #676767 |   |   |  
1.4.11 Non-Text Contrast | AA | Fail | Interface contrast |   |   |   | Easy |   |   | Interactive controls don't have a border or any identifier that passes the minimum contrast requiremnets of 3:1 with the background |   |   | https://jolevesq.github.io/contributed-plugins/thematic-slider/samples/thematic-slider-index.html?sample=9 |   |   | Solution 1: Make the buttons and icons darker. Make it at least #929292. |   |   |  
1.4.11 Non-text Contrast (WCAG 2.1) | AA | Fail | Interactive Content | Yes |   |   | Moderate |   |   | Focus ring (dotted/solid line around the control that has focus) has insufficient contrast around certain interactive controls as user tabs through the page. | Interactive elements do not show visual focus when it receives tab focus. | A keyboard user tabbing through the page will not know when focus is on controls that don't have a focus ring. | https://jolevesq.github.io/contributed-plugins/draw/samples/draw-index.html |   |   | The anchor outline:none appears to over ride the :focus class to remove the visual focus when the tab key is used to access the element.. To solve this, remove outline:none from the anchor classAnd focus:noneNote: if these are user agent Style sheet settings, simply override them with visible outline values in the CSSIn the example solution to the left its a grey outline. It could be and dark color such as red, dark blue, green etc... |   | Delete this {outline: none} This code will ensure the focus is always visible and has suffcient contrast regardless of the background.*:focus { box-shadow: inset 0 0 0 1px black, 0 0 3px 1px white !important;outline: dashed !important;}Live example linked to the right |  
1.4.11 Non-text Contrast (WCAG 2.1) | AA | Fail | Interactive Content | Yes |   |   | Moderate |   |   | For the graphing maps, the arrow keys have a very coarse adjustment and it is difficult to set focus on the small dots which don;t increase in size when the user zooms in. |   |   | https://jolevesq.github.io/contributed-plugins/draw/samples/draw-index.html |   |   | Provide a modulating key which makes the cursor move much more granular. I suggest trapping the CONTROL key and when the user holds it down and presses the arrows the cursor moves much more granualarly, (slower).There could also be a modulating key to make it go faster. For instance, when the ALT key is helod sown, pressing the arrow keys is much less granular and the cursor moves quicker. |   |   |  
1.4.3 Contrast (Minimum) | AA | Fail | Colour or Contrast |   |   |   | Moderate |   |   | Text (or images of text) that is <18 point (24px) or 14 point (18.5px) bold does not have a colour contrast of 4.5:1 or greater (logos exempt) OR there is no global mechanism (top right) that allows users to switch to high contrast mode |   | People with low vision and colour blindness may have trouble seeing it. It's difficult to see content that doesn't have sufficient contrast. |   |   |   | Make the text at least #676767 |   | Consider a slightly darker grey text. This is a threshold grey. #767676 | A global strategy to address the colour contrast issues may be to provide a high contrast icon on the top right corner of the site which will make the site easier to use for people with low vision. A free contrast tool can be downloaded. Google "Color Contrast Tool" (British spelling of color)
2.1.1 Keyboard | A | BP | Button | Yes | Yes |   |   |   |   | Hamburger menu stays open if an item is activated with mouse but closes if a keyboard is used.  With mouse sometimes it stays open sometimes it stays closed. |   |   | https://jolevesq.github.io/contributed-plugins/thematic-slider/samples/thematic-slider-index.html?sample=99 |   |   | Leave the side panel open until the user dismisses it with "close" or perhaps an OK button.  If ia button on the side panel opens a dialog, then the normal behaviour is to return the user to the button that launched the dialog when the dialog is closed. An example would be the export button which opens a dialog box and closes the side menu, but doesn't return the user back ato the export putton when the dialog is closed. A more natural behavipur would be to leave the side panel open when launching the dialog (it can be greyed out with the rest of the background.) Then return the user to the export button when the dialog is closed. |   |   |  
2.1.1 Keyboard | A | Fail | Dialog | Yes | Yes |   |   |   |   | The mechanism to go into keyboard mode is visually presented as a dialog box, with a greyed out background. But is doesn't act like one and doesn't present the proper roles and states. |   |   | https://jolevesq.github.io/contributed-plugins/draw/samples/draw-index.html |   |   | There are a number of fixes1) Don't have it automatically popup. That is confusing.2) Provide an invisible button that shows up on focus that says "Open keyboard Instructions"3) This would open a a true dialog, where the focus is placed inside it4) Tab focus would remain inside it, and it is dismised either by hitting then OK button, the X button or pressing escape ESC key.5) a model for the modal is to the right. It provides the HTML, CSS, JavaScript and ARIAThe instructions could then be more comprehensive to describe how to draw on the map etc. if that is aporopriate for the functionality of the map. |   | https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html |  
2.1.1 Keyboard | A | BP | Button | Yes | Yes |   |   |   |   | When drawing lines with the keyboard (like echisketch with the arrows) The tooltip says, press space to end line. When the line is ended, the user is not returned to the control that activated the tool. They are sent to the menu. |   |   | https://jolevesq.github.io/contributed-plugins/draw/samples/draw-index.html |   |   | When the spacebar is pressed to end the line, move the keyboard focus back to the button that activated the line. |   |   |  
2.1.1 Keyboard | A | BP | Button | Yes | Yes |   |   |   |   | The "edit existing drawing" tool doesn't work with keyboard. |   |   | https://jolevesq.github.io/contributed-plugins/draw/samples/draw-index.html |   |   | The short term solution be to open an alert dialog box if the button is activated with the ENTER key (Keycode 13) and notify the user that this functionality is not providedfor keyboard because of the complexity of the functionality.A longer term solution (or maybe short) could be to allow the keyboard user to draw boxes and hit the delete key to remove any lines in the boxes. In that case when the user hits Edit, the dialog would ooen to inform them of this functionality and dismissing the dialog box would take them to the map to start. |   |   |  
4.1.2 Name, Role, Value | A | Fail | Modal |   |   |   |   |   |   | Modal dialog is missing its roles states and name |   | When the side panel is open the background is greyed out visually which makes it a modal. It should announce as such and should keep keyboard focus inside it. |   |   |   | add to the dialog container  role="dialog" aria-modal="true" aria-labelledby="[id of heading for dialog]" |   | <md-sidenav... role="dialog"  aria-modal="true"  aria-labelledby="p1"><h2 id="p1">Modal heading text</h2>... | https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html
4.1.2 Name, Role, Value | A | Fail | Modal |   |   |   |   |   |   | There is a long/lat cordinates provided on the screen to provide the location of the mouse, butnthis is not provided to the keyboard user. |   |   | https://jolevesq.github.io/contributed-plugins/draw/samples/draw-index.html |   |   | Provide the same functionality as the cursor is moved around with the keyboard. |   |   |  
1.4.1 Use of Color | A | Fail | Colour or Contrast |   |   |   | Easy |   |   | No other means given to identify content besides the colour. |   | The legend and map don't have an alternative way to present the data, except in color |   |   |   |   |   |   |
